code/game/objects/items/weapons/RCD.dm

```diff
@@ -563,12 +581,16 @@
 	switch(mode)
 		if(MODE_TURF)
 			. = mode_turf(A, user)
+			update_icon()
 		if(MODE_AIRLOCK)
 			. = mode_airlock(A, user)
+			update_icon()
 		if(MODE_DECON)
 			. = mode_decon(A, user)
+			update_icon()
 		if(MODE_WINDOW)
 			. = mode_window(A, user)
+			update_icon()
 		else
```

Comment:
These can be just once following the switch

Reply:
i have monkey brain. this is fixed